iwyu_output.cc

```diff
@@ -1426,6 +1426,71 @@ void CalculateIwyuForFullUse(OneUse* use,
   }
 }
 
+// This function removes all include lines `a` that are desired but not present but
+// transitively included by some include line `b` that is desired and present and
+// adds `a`'s symbol counts to `b`.
+vector<OneIncludeOrForwardDeclareLine> PruneTransitivelyPresent(
+    const vector<OneIncludeOrForwardDeclareLine>& lines,
+    const IwyuPreprocessorInfo* preprocessor_info) {
+  // In the first step, we copy all lines that are an include line, not present,
+  // desired, and whose include_file is transitively included by another line
+  // which is a desired include line into a range `indirect` and the remaining
+  // ones into another range `direct`. This is because elements of range
+  // `indirect` are then precisely those whose symbols are already exposed
+  // through a different header that is to be included anyway, so they can be
+  // omitted.
+  const auto is_transitively_present =
+      [&lines,
+       preprocessor_info](const OneIncludeOrForwardDeclareLine& target) {
+        if (!target.IsIncludeLine() || target.is_present() ||
```

Comment:
`is_present` and `is_desired` are both cheaper than `IsIncludeLine`, so please put them first.